viewer/packages/tools/src/Toolbar/Toolbar.ts

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import { Cognite3DViewer } from '@reveal/core';
+import { assertNever } from '@reveal/utilities';
+import css from './Toolbar.css';
+
+/**
+ * Toolbar position within the canvas
+ */
+export type ToolbarPosition = 'top' | 'right' | 'bottom' | 'left';
+
+/**
+ * Toolbar class which creates a container to hold all icons within the toolbar
+ */
+export class Toolbar {
+  private _toolbarContainer: HTMLDivElement;
+  private static readonly stylesId = 'reveal-viewer-toolbar-styles';
+  private readonly _canvasElement: HTMLCanvasElement | null;
+
+  private static readonly classnames = {
+    container: 'reveal-viewer-toolbar-container',
+    bottom: 'reveal-viewer-toolbar-container--bottom',
+    top: 'reveal-viewer-toolbar-container--top',
+    left: 'reveal-viewer-toolbar-container--left',
+    right: 'reveal-viewer-toolbar-container--right',
+    icon: 'reveal-viewer-toolbar-icon',
+    iconImg: 'reveal-viewer-toolbar-icon-img',
+    iconClicked: 'reveal-viewer-toolbar-icon-clicked'
+  };
+
+  constructor(viewer: Cognite3DViewer) {
+    this._canvasElement = viewer.domElement.querySelector('canvas');
+    const canvasElementParent = this._canvasElement?.parentElement;
+    if (canvasElementParent === null) {
+      throw new Error('Could not find canvas');
+    }
+
+    this._toolbarContainer = document.createElement('div');
+    this.createToolBar(canvasElementParent!);
+  }
+
+  /**
+   * Load the the styles from the CSS and appends them to the toolbar
+   * @returns Return if styles already loaded
+   */
+  private static ensureStylesLoaded() {
+    if (document.getElementById(Toolbar.stylesId)) {
+      return;
+    }
+
+    const style = document.createElement('style');
+    style.id = Toolbar.stylesId;
+    style.appendChild(document.createTextNode(css));
+    document.head.appendChild(style);
+  }
+
+  /**
+   * Creates Toolbar container and adds it to the Canvas parent element and assigns default styles to them
+   * @param controlDiv Canvas parent div element
+   */
+  private createToolBar(controlDiv: HTMLElement) {
+    this._toolbarContainer.id = 'toolbarContainer';
+    Toolbar.ensureStylesLoaded();
+    this._toolbarContainer.className = Toolbar.classnames.container;
+    this._toolbarContainer.classList.add(Toolbar.classnames.bottom);
+
+    controlDiv.appendChild(this._toolbarContainer);
+  }
+
+  /**
+   * Create & adds a button icons into the toolbar container
+   * @param toolTip Tooltip message to added for the icon
+   * @param backgroundImageUri Icon image to be displayed
+   * @param isToggle Is the icon button used as toggle
+   * @param onClick Click event action for the icon
+   */
+  public addToolbarItem(toolTip: string, backgroundImageUri: string, isToggle: boolean, onClick: () => void): void {
```

Comment:
I think it would make sense to have `addToolbarButton` and `addToolbarToogleButton`. The signature for `addToolbarToogleButton` can be e.g.
```
addToolbarToogleButton(backgroundImageUri: string, isToggle: boolean, onToggled: (isToggled: boolean) => void, toolTip: string = '');
```

Also, add tooltip as the last argument and make it optional.